Builtin Vite support for backend

haikyuu opened the pull request.

somebee commented on packages/imba/src/imba/setupVite.imba:
```diff
@@ -0,0 +1,47 @@
+# imba$stdlib=1
+import np from 'path'
+import * as Vite from "vite"
```
Won't this end up importing vite whenever you run imba in node? Or do you know if it is removed by tree-shaking in esbuild?

haikyuu replied:
Sorry for the late reply 🙂 There seem to be issues related to this 

Running a script complains about
```
❯ imba index.imba
▲ [WARNING] "esbuild" should be marked as external for use with "require.resolve" [require-resolve-not-external]

    ../imba/packages/imba/node_modules/esbuild/lib/main.js:1718:36:
      1718 │   const libMainJS = require.resolve("esbuild");
           ╵                                     ~~~~~~~~~

✘ [ERROR] No loader is configured for ".node" files: ../imba/packages/imba/node_modules/fsevents/fsevents.node

    ../imba/packages/imba/node_modules/fsevents/fsevents.js:13:23:
      13 │ const Native = require("./fsevents.node");
         ╵                        ~~~~~~~~~~~~~~~~~

× failed with 1 errors
```

haikyuu replied:
I removed `setupVite` from imba directly into `imba/plugin`. It was indeed causing the issue :) Nice catch

haikyuu commented:
Rebased with master and fixed a bug where it fails if no tsconfig file is found (depends on this PR upstream https://github.com/aleclarson/vite-tsconfig-paths/pull/97)

haikyuu commented:
Here is a list of things I consider missing from Vite support to get out of experimental:
- [ ] Make sure Vite and its other dependencies don't affect people using the built-in bundler
- [ ] Fix bug in this PR in above comment
- [ ] Switch to a single `imba.config.js` file for configuring the bundling of client, server and tests
- [ ] Out of the box (no config needed) vite xp for client only, SSR and tests

haikyuu commented:
rebased with master to have a green CI
